### PR TITLE
feat: manta wallet UI integration

### PR DESCRIPTION
**Pull request by jasonintju**
- [UI] manta wallet intro modal
  image
- [UI] toggle between Manta Signer and Manta Wallet
  image
- [UI] update IP blocking modal
  image
- [UI] update wallet connect modal
  image
- [logic] define a global state: usingMantaWallet

### CODE REVIEW

**Review by Kevingislason — COMMENTED** [review]
Code looks great aside from my three tiny comments. Nice work!

Some UI tweaks: 
- In the Connect Wallet modal, there is a bit too much space beneath "Manta Signer user?
[Learn how to migrate from Manta Signer to Manta Wallet"
- On the "Manta Wallet Extension is Live" modal, I don't see the text "Manta Signer user? And still want to use it?" at the bottom. The text "Don't show this message again" should also be to the right.

The one big remaining thing to do is to show both zkAddress and public address when MantaWallet is active (see image), but you don't need to do that in this PR. 

Screen Shot 2023-04-06 at 6 23 04 PM

File: src/pages/SendPage/SendForm.tsx [hunk 1]
```diff
@@ -30,6 +32,12 @@ const SendForm = () => {
     }
   }, [keyring]);
 
+  const toggleUsingMantaWalletState = () => {
+    // toggle global status
+    const newState = !usingMantaWallet;
+    setUsingMantaWallet(newState);
```
[the remark this answers is not in the document]
Comment (jasonintju) [reply src/pages/SendPage/SendForm.tsx:38]: Yeah, absolutely. 
In fact at first, I wanna use `setUsingMantaWallet(usingMantaWallet => !usingMantaWallet)`, which is a recommended practice, but then I realized the param must be a type of `boolean`, so I just simply picked out the value. Haha, it's funny, it should have been in one line. 😃

File: src/components/Modal/connectWalletModal.tsx [hunk 2]
```diff
@@ -145,6 +147,50 @@ const MetamaskConnectWalletBlock = ({hideModal}) => {
   );
 };
 
+const MantaConnectWalletBlock = ({ setIsMetamaskSelected, hideModal }) => {
+  // TODO
+  // get manta wallet meta data
+  // { extensionName, installed, installUrl, logo, isWalletEnabled }
```
[the remark this answers is not in the document]
Comment (jasonintju) [reply src/components/Modal/connectWalletModal.tsx:153]: yeah agree

**Comment by jasonintju** [comment]
> * On the "Manta Wallet Extension is Live" modal, I don't see the text "Manta Signer user? And still want to use it?" at the bottom. The text "Don't show this message again" should also be to the right.

look at the latest version of mockup here https://www.figma.com/file/9nAYHsYlfHmyT8RL5P8ISZ/Manta-Pay?node-id=4390-5790&t=gCBoNdxrZsgcVZxk-0, some ui has changed.

**Comment by jasonintju** [comment]
> The one big remaining thing to do is to show both zkAddress and public address when MantaWallet is active (see image), but you don't need to do that in this PR.

Yeah, I didn't consider this as a part of UI jobs, I think it's almost all about logic. I will implement it in another PR.

**Review by jincan39 — APPROVED** [review]

**Review by Kevingislason — APPROVED** [review]

**Pull request merged**